### ISSUE TITLE
karma-coverage-istanbul-reporter author requires payment for support and issue reporting

### ISSUE THREAD

**Issue by muuvmuuv** [angular/angular-cli#17757]
# 🐞 Bug report

### Command (mark with an `x`)

- [x] new

### Is this a regression?

I don't know but I think it must have worked.

### Description

I have installed a starter through `ng new` leaved everything the default and tried to create a coverage report by enabling `karma-coverage-istanbul-reporter` with `reporters: ['progress', 'coverage-istanbul', 'kjhtml'],`. This is how the author describes the installation on his repo. Sadly this does not work and just prints

```
Statements   : Unknown% ( 0/0 )
Branches     : Unknown% ( 0/0 )
Functions    : Unknown% ( 0/0 )
Lines        : Unknown% ( 0/0 )
```

I tried to fill an issue but this author **REALLY WANTS $100/month** for that!!! This is in acceptable for an OSS project despite the fact that it is coupled to Angular. This makes me angry because new developer will see this and will get a wrong definition of what OSS is and maybe follow this behaviour. Sure, include a sponsor button and say that you don't want to work on it without receiving a _SMALL_ amount of money but then add a "help wanted" badge and ask for help or use issuehunt. I would love to help to solve this issue but nope, [my issue gets closed and locked](https://github.com/mattlewis92/karma-coverage-istanbul-reporter/issues/83) (like many others).

I suggest removing that reporter from the starter and let the author know that this isn't a behaviour Angular supports.

## 🔬 Minimal Reproduction

Fresh starter with coverage enabled.

## 🔥 Exception or Error

<pre><code>
Statements   : Unknown% ( 0/0 )
Branches     : Unknown% ( 0/0 )
Functions    : Unknown% ( 0/0 )
Lines        : Unknown% ( 0/0 )
</code></pre>

## 🌍 Your Environment

<pre><code>
Angular CLI: 9.1.6
Node: 12.16.1
OS: darwin x64

Angular: 9.1.7
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.6
@angular-devkit/build-angular     0.901.6
@angular-devkit/build-optimizer   0.901.6
@angular-devkit/build-webpack     0.901.6
@angular-devkit/core              9.1.6
@angular-devkit/schematics        9.1.6
@angular/cli                      9.1.6
@ngtools/webpack                  9.1.6
@schematics/angular               9.1.6
@schematics/update                0.901.6
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0
</code></pre>

**Anything else relevant?**

I don't know which coverage reporter I should use now...

**Comment by muuvmuuv** [comment]
I forgot to use `--code-coverage` but anyway this doesn't change the way he forces users to pay for support. I cannot add this solution with `--code-coverage` now for other users because it is locked. Same question got asked on SO. And yes I should have searched the NG docs before ^^ sorry for that.

**Comment by Splaktar** [comment]
Updated the title to reflect the fact that the plugin is actually working if used as [documented](https://angular.io/guide/testing#enable-code-coverage-reports).

**Comment by MuthuD** [comment]
I'm also facing this **Unknown%** issue when upgraded from Angular 6 to 9. 
And I'm using --code-coverage, still not getting any coverage output.
Can someone guide me on this?

Are we going to change karma-coverage-Istanbul-reporter to karma-coverage?

**Comment by mattlewis92** [comment]
Hey there, I just spotted this issue while looking for something else and thought I should offer an explanation. I've been writing and maintaining open source in my free time for the last 5 years and never earned a penny from it until earlier this year, it's extremely exhausting working on something in your spare time for free, especially when you know that the majority of your users are using it for profit. Charging for support on some of my repos was an experiment to see if I could make maintenance less of a burden, but I can now see that it wasn't a good idea, and I've taken it off in response to this issue as I think it causes more harm than good. I've got some better ideas around open source monetisation that are a bit less abrasive and work better to satisfy the needs of the community. Thanks for raising this issue and I really do appreciate the feedback ✌️
